client/verta/verta/_internal_utils/_utils.py

```diff
@@ -259,19 +260,53 @@ def make_request(method, url, conn, **kwargs):
     with requests.Session() as s:
         s.mount(url, HTTPAdapter(max_retries=conn.retry))
         try:
-            response = s.request(method, url, **kwargs)
+            request = requests.Request(method, url, **kwargs).prepare()
+            response = s.send(request, allow_redirects=False)
+
+            # manually inspect initial response and subsequent redirects to stop on 302s
+            history = []  # track history because `requests` doesn't since we're redirecting manually
+            responses = itertools.chain([response], s.resolve_redirects(response, request))
+            for response in responses:
+                if response.status_code == 302:
+                    if not conn.ignore_conn_err:
+                        raise RuntimeError(
+                            "received status 302 from {},"
+                            " which is not supported by the Client".format(response.url)
+                        )
+                    else:
+                        return fabricate_200()
```

Comment:
Why would we fabricate a 200 on a 302? If the user wants to ignore errors, shouldn't we just leave it be and return the original response?

Reply:
For context, the way that the Client's `ignore_conn_err` flag is intended to function is that this util returns a non-error HTTP response with an empty JSON object in its body. Then when the body is loaded into protobuf objects, the message's fields will be populated with their default values. So the Client can continue to execute as much as possible even if it's receiving `4XX`s, `5XX`s, or connection errors.

So for me, the key part here as that empty JSON body, because it enables that default protobuf object behavior in the caller. The original `302` might not have that, but `fabricate_200()` does.

Alternatively, I could reassign the content of the original `302`.

Reply:
Ok